lib/backpex/adapters/ash.ex

```diff
@@ -0,0 +1,113 @@
+defmodule Backpex.Adapters.Ash do
+  @config_schema [
+    resource: [
+      doc: "The `Ash.Resource` that will be used to perform CRUD operations.",
+      type: :atom,
+      required: true
+    ]
+  ]
+
+  @moduledoc """
+  The `Backpex.Adapter` to connect your `Backpex.LiveResource` to an `Ash.Resource`.
+
+  ## `adapter_config`
+
+  #{NimbleOptions.docs(@config_schema)}
+
+  > ### Work in progress {: .error}
+  >
+  > The `Backpex.Adapters.Ash` is currently not usable! It can barely list and show items. We will work on this as we continue to implement  the `Backpex.Adapter` pattern throughout the codebase.
+  """
+
+  use Backpex.Adapter, config_schema: @config_schema
+  require Ash.Query
+
+  @doc """
+  Gets a database record with the given primary key value.
+
+  Returns `nil` if no result was found.
+  """
+  @impl Backpex.Adapter
+  def get(primary_value, _assigns, live_resource) do
+    config = live_resource.config(:adapter_config)
+    primary_key = live_resource.config(:primary_key)
+
+    config[:resource]
+    |> Ash.Query.filter(^Ash.Expr.ref(primary_key) == ^primary_value)
+    |> Ash.read_one()
+  end
+
+  @doc """
+  Gets a database record with the given primary key value.
+
+  Raises `Ecto.NoResultsError` if no record was found.
+  """
+  @impl Backpex.Adapter
+  def get!(primary_value, _assigns, live_resource) do
+    config = live_resource.config(:adapter_config)
+    primary_key = live_resource.config(:primary_key)
+
+    config[:resource]
+    |> Ash.Query.filter(^Ash.Expr.ref(primary_key) == ^primary_value)
+    |> Ash.read_one!()
+  end
+
+  @doc """
+  Returns a list of items by given criteria.
+  """
+  @impl Backpex.Adapter
+  def list(_criteria \\ [], _fields, _assigns, config) do
+    config[:resource]
+    |> Ash.read!()
+  end
+
+  @doc """
+
+  """
+  @impl Backpex.Adapter
+  def count(_criteria \\ [], _fields, _assigns, config) do
+    config[:resource]
+    |> Ash.count!()
+  end
+
+  @doc """
+  Deletes multiple items.
+  """
+  @impl Backpex.Adapter
+  def delete_all(items, _config) do
+    items
+    |> Ash.bulk_destroy!(:destroy, %{})
+  end
```

Comment:
`Backpex.Adapter` expects `{:ok, term()} | {:error, term()}` as a result of `delete_all/2`, but this call returns `Ash.BulkResult.t() | no_return()`

Reply:
I removed the code and added `raise "not implemented yet"`